mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_deserializer.rs

```diff
@@ -171,10 +192,32 @@ mod tests {
             id: &str,
         ) -> Result<SoftwareUpdateRequest, SmartRestDeserializerError> {
             let request = SoftwareUpdateRequest::new_with_id(id);
-            Ok(self.map_to_software_update_request(request)?)
+            self.map_to_software_update_request(request)
         }
     }
 
+    #[test]
+    fn jwt_token_create_new() {
+        let jwt = SmartRestJwtResponse::new();
+
+        assert!(jwt.token.is_empty());
+    }
+
+    #[test]
+    fn jwt_token_deserialize() {
+        let test_response = "71,123456";
+        let mut csv = csv::ReaderBuilder::new()
+            .has_headers(false)
+            .from_reader(test_response.as_bytes());
+
+        let mut jwt = SmartRestJwtResponse::new();
+        for result in csv.deserialize() {
+            jwt = result.unwrap();
+        }
+
+        assert_eq!(jwt.token(), "123456");
+    }
+
```

Comment:
These tests have little value (testing serde csv not the code). Furthermore the test code (building a csv reader and deserialize the string sent by c8y) is duplicated in the mapper. It must be encapsulated.

I would replace the `SmartRestJwtResponse::new()` function with a `maybe_new(input: &str)` that does the parsing, validate the 71 code and return the JwtToken if any.

The test would be then:

```rust
    #[test]
    fn jwt_token_deserialize() {
        let jwt = SmartRestJwtResponse::maybe_new("71,123456").unwrap();
        assert_eq!(jwt.token(), "123456");
    }
```